packages/core/src/components/overlay/overlay.tsx

```diff
@@ -159,6 +159,19 @@ export class Overlay extends React.Component<IOverlayProps, IOverlayState> {
 
         const { children, className, inline, isOpen, transitionDuration, transitionName } = this.props;
 
+        // add a special class to each child that will automatically set the appropriate
+        // CSS position mode under the hood. we add this class *after* the existing classes
+        // so that our class will override the position style of other classes that might
+        // have set unnecessary position rules (note that CSS specificity plays a role here;
+        // our class will only override an existing position rule if it has the same or
+        // greater selector specificity).
```

Comment:
If we proceed with this approach, might be worth mentioning in the docs that there's no need to explicitly set `position` on your overlay content container, as we manage that ourselves.